src/PowerShellEditorServices/Session/PowerShellContext.cs

```diff
@@ -796,21 +798,22 @@ public async Task ExecuteScriptWithArgs(string script, string arguments = null,
                 if (File.Exists(script) || File.Exists(scriptAbsPath))
                 {
                     // Dot-source the launched script path
-                    script = ". " + EscapePath(script, escapeSpaces: true);
+                    escapedScriptPath = ". " + QuoteEscapeString(escapedScriptPath);
                 }
 
-                launchedScript = script + " " + arguments;
-                command.AddScript(launchedScript, false);
+                reportedScript += " " + arguments;
+                escapedScriptPath += " " + arguments;
```

Comment:
Can you use a StringBuilder or string interpolation here instead?

Same for line 804, 801